lib/controllers/PrometheusController.ts

```diff
@@ -1,117 +1,64 @@
- * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
- * See the License for the specific language governing permissions and
- * limitations under the License.
- */
-'use strict';
+import {
+  ControllerDefinition,
+  PluginContext,
+  KuzzleRequest,
+  JSONObject,
+} from 'kuzzle';
+
+import os from 'os';
+import Prometheus from 'prom-client';
+import util from 'util';
+import macaddress from 'macaddress';
 
-const
-  os = require('os'),
-  Prometheus = require('prom-client');
-
-/**
- * @class PrometheusPlugin
- *
- * @property {KuzzlePluginContext}     context        Kuzzle plugin context
- * @property {Object.<string, string>} hooks          Kuzzle plugin hooks
- * @property {Routes}                  routes         Kuzzle plugin routes
- * @property {Controllers}             controllers    Kuzzle plugin controllers
- * @property {Prometheus.Registry}     registry       Prometheus Registry
- * @property {Object.<string, Object>} kuzzleMetrics  Prometheus metrics dedicated for Kuzzle
- * @property {String}                  key            Key used when pushing Prometheus Registry to Redis
- * @property {String}                  hostname       Hostname
- * @property {String}                  MAC            Host MAC address
- * @property {String}                  IP             Host IPv4 adress
- *
- * @externs
- */
-class PrometheusPlugin {
-  /**
-   * PrometheusPlugin constructor
-   *
-   * @returns {PrometheusPlugin}
-   */
-  constructor() {
-    this.config = {
-      collectSystemMetrics: true,
-      systemMetricsInterval: 5000,
-      labels: {
-        common: [
-          'nodeHost',
-          'nodeMAC',
-          'nodeIP'
-        ],
-        kuzzle: [
-          'controller',
-          'action',
-          'event',
-          'protocol',
-          'status'
-        ]
-      }
-    };
-
-    this.controllers = {
-      prometheus: {
-        metrics: 'metrics'
-      }
-    };
-
-    this.routes = [
-      {
-        verb: 'get',
-        url: '/metrics',
-        controller: 'prometheus',
-        action: 'metrics'
+const getIPFromMAC = mac => {
+  const interfaces : NodeJS.Dict<os.NetworkInterfaceInfo[]> = os.networkInterfaces();
+  for (const inet of Object.keys(interfaces)) {
+    for (const settings of interfaces[inet]) {
+      if (settings.mac === mac) {
+        return settings.address;
       }
-    ];
-
-    this.hostname = os.hostname();
-
-    this._mac = null;
-    this._ip = null;
+    }
   }
+};
 
+export class PrometheusController {
+  private context: PluginContext;
+  private config: JSONObject;
 
-  get MAC() {
-    return this._mac;
-  }
+  private _mac? : string;
+  private _ip? : string;
+  private _hostname? : string;
+  private _registry? : Prometheus.Registry;
+  private _kuzzleMetrics? : JSONObject;
+  private _systemMetricsJob? : any;
 
-  get IP() {
-    return this._ip;
-  }
+  public definition: ControllerDefinition;
+  public hooks: JSONObject;
```

Comment:
What is this used for ?

Reply:
Actually it looks like we never add a hook, if this is the case then this property is useless.

Also, in general it's not up to a controller class to declare hooks